financial-templates-lib/price-feed/CreatePriceFeed.js

```diff
@@ -147,6 +172,14 @@ async function getUniswapPairDetails(web3, syntheticTokenAddress, collateralCurr
   return {};
 }
 
+async function createBalancerPriceFeedForEmp(logger, web3, networker, getTime, empAddress, config = {}) {
+  assert(empAddress, "createUniswapPriceFeedForEmp: Must pass in an `empAddress`");
+  const emp = getEmpAtAddress(web3, empAddress);
+  const balancerTokenOut = await emp.methods.collateralCurrency().call();
+  const balancerTokenIn = await emp.methods.tokenCurrency().call();
+  return createPriceFeed(logger, web3, networker, getTime, { balancerTokenIn, balancerTokenOut, ...config });
```

Comment:
Also if you want to find the balancer pool address & you know the tokens within the pool you can run a query like this on thegraph:

```
{
      pools(where: {tokensList: ["0x81ab848898b5ffd3354dbbefb333d5d183eedcb5","0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"]}) {
        id
      }
    }
```

This will return something like this:
```
{
  "data": {
    "pools": [
      {
        "id": "0x58ef3abab72c6c365d4d0d8a70039752b9f32bc9"
      }
    ]
  }
}
```

which contains the pool address. however here you _need_ to know both the synthetic and reference tokens. I suspect it is possible to construct a query that will give you the pool with only the synthetic token address but I'm not 100% sure what it would look like.

Reply:
I think in general for the bots we should stick to contract calls as they are more reliably than subgraph data, especially because this pricefeed will be used by the liquidator and disputer bots.